sdv/metadata/multi_table.py

```diff
@@ -354,11 +370,33 @@ def detect_table_from_csv(self, table_name, filepath):
         """
         self._validate_table_not_detected(table_name)
         table = SingleTableMetadata()
-        data = table._load_data_from_csv(filepath)
-        table._detect_columns(data)
+        table.detect_from_csv(filepath)
         self.tables[table_name] = table
         self._log_detected_table(table)
 
+    def detect_from_csvs(self, folder_name):
+        """Detect the metadata for all tables in a folder of csv files.
+
+        Args:
+            folder_name (str):
+                Name of the folder to detect the metadata from.
+
+        """
+        if os.path.exists(folder_name) and os.path.isdir(folder_name):
+            csv_files = [
+                filename for filename in os.listdir(folder_name) if filename.endswith('.csv')
+            ]
```

Comment:
Instead of using `os.path` we can use `Pathlib` which is more powerful and will improve the code down the line.
For example:
```python
Path().rglob(f'{folder_name}/*.csv')
```
will return a list with all the paths of csv files-

Reply:
This is really good advice, thanks! Done in [4c37e19](https://github.com/sdv-dev/SDV/pull/1533/commits/4c37e19e8e25d340c6c5693b237d1e46fe0d9c28).